### ISSUE TITLE
False positive for PossiblyNullReference

### ISSUE THREAD

**Issue by theseer** [vimeo/psalm#4324]
I ran into this while working on an Event Sourced system.

Not sure if it really qualifies as a bug or false positive, but at least this was a bit of a surprise.

When calling a private method that is never called without having an initializing method first, the use of the nullable property should not be considered a "PossiblyNullReference".

The use call of `$this->x->a();` within the `test` method is accepted, due to the `psalm-assert` annotation for `init`. But the very same annotation should - imho - make psalm happy for the use within the `cleanup` method as that method is never called without having `init` done before hand.

The method `cleanup` is private, so no external calls can be made. Within the class, the only call to it  is from within `test`. Which means, the fact that `init` was called before should ensure that the property is set and can not be `null` at the time `cleanup` is called-.

```php
<?php
class bar {
	public function a(): void {}
}

class foo {
 
    private ?bar $x = null;
    
    public function test(): void {
        $this->init();    
		$this->x->a();    
        $this->cleanup();
    }
    
    /** @psalm-assert bar $this->x */
    private function init(): void {
        $this->x = new bar();
    }
    
    private function cleanup(): void {
        $this->x->a();
    }
}
```

Tried via https://psalm.dev/r/a2cde3b52f

**Comment by muglug** [comment]
This is as I'd expect, though you can pass `$this->x` to `cleanup`: https://psalm.dev/r/c42369e8af
